### PR TITLE
Bumped drupal version

### LINKED ISSUE #222: Drupal 10 readiness
## Is your feature request related to a problem? Please describe.
Bump Drupal core version and dependency to support Drupal 10.

### PR DESCRIPTION
**Pull request by divya-intelli**
Fixes #222

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -4,16 +4,16 @@
     "type": "drupal-module",
     "description": "Apigee API Catalog for Drupal",
     "require": {
-        "php": "^8.0",
+        "php": "~8.0 || ~8.1",
         "drupal/entity": "^1.1",
         "drupal/file_link": "^2.0",
-        "drupal/apigee_edge": "^2.0",
+        "drupal/apigee_edge": "^2.0 || dev-drupal10-prep",
```
Comment (giteshk) [inline composer.json]: 3.x

File: .github/workflows/php.yml [hunk 2]
```diff
@@ -35,6 +35,11 @@ jobs:
           # Should update the following as the minimum supported version from Drupal.org
           - "9.4.x"
           - "9.5.x"
+        include:
+          - php-version: "8.1"
+            drupal-core: "10.0.x"
+          - php-version: "8.1"
+            drupal-core: "10.1.x"
```
Comment (giteshk) [inline .github/workflows/php.yml:43]: lets add PHP 8.2, even if it fails

File: composer.json (hunk 1 above)
Comment (divya-intelli) [reply composer.json]: added edge version 3.0.0

File: .github/workflows/php.yml (hunk 2 above)
Comment (divya-intelli) [reply .github/workflows/php.yml:43]: Added ticket for this #227 227

**Review by shishir-intelli — APPROVED** [review]
LGTM

**Pull request merged**